runtime/src/main/java/org/corfudb/runtime/view/SequencerView.java

```diff
@@ -20,26 +21,44 @@ public SequencerView(CorfuRuntime runtime) {
     }
 
     /**
-     * Return the next token in the sequencer for a particular stream.
+     * Return the next token in the sequencer for the global tail or the tails
+     * of multiple streams.
      *
-     * <p>If numTokens == 0, then the streamAddressesMap returned is the last handed out token for
-     * each stream (if streamIDs is not empty). The token returned is the global address as
-     * previously defined, namely, max global address across all the streams.</p>
+     * @param streamIds the streams to query
+     * @return the global tail or a list of tails
+     */
+    public TokenResponse queryTails(UUID ... streamIds) {
+        if (streamIds.length == 0) {
+            return layoutHelper(e -> CFUtils.getUninterruptibly(e.getPrimarySequencerClient()
+                    .nextToken(Collections.emptyList(), 0)));
+        } else {
+            return layoutHelper(e -> CFUtils.getUninterruptibly(e.getPrimarySequencerClient()
+                    .nextToken(Arrays.asList(streamIds), 0)));
+        }
+    }
+
+    /**
+     * Return the next token in the sequencer for a particular stream.
      *
-     * @param streamIDs The stream IDs to retrieve from.
-     * @param numTokens The number of tokens to reserve.
+     * @param streamIds The stream IDs to retrieve from.
      * @return The first token retrieved.
      */
-    public TokenResponse nextToken(Set<UUID> streamIDs, int numTokens) {
+    public TokenResponse nextToken(UUID ... streamIds) {
```

Comment:
If you're going to change the API signature, I would suggest renaming it to next(). Also, please retain the legacy API but mark it deprecated.

Reply:
Done.